### PR TITLE
Update to latest revault_tx followups

### PR DESCRIPTION
**Pull request by danielabrozzoni**
Rebase of #217 + some fixes

~~TODO: tests :)~~

### CODE REVIEW

**Comment by darosior** [comment]
Cool! Will review as soon as it passes CI :)

**Comment by danielabrozzoni** [comment]
Should be good now, I just need to fix the Cargo.toml once revault_net and revault_tx are tagged

**Comment by darosior** [comment]
Functional tests are failing

**Comment by danielabrozzoni** [comment]
Timeout on `activate_fresh_vaults`. Again. FFS :sob:

File: src/daemon/control.rs [hunk 1]
```diff
@@ -508,7 +508,7 @@ pub fn presigned_tx_sighash(
     hashtype: SigHashType,
 ) -> Result<secp256k1::Message, SigError> {
     // Presigned transactions only have one input when handled by revaultd.
-    if !tx.tx().input.len() == 1 {
+    if tx.tx().input.len() != 1 {
```
Comment (darosior) [inline src/daemon/control.rs:511]: Wtf wasn't this part of master already?

File: src/daemon/sigfetcher.rs [hunk 2]
```diff
@@ -114,13 +111,7 @@ fn get_sigs(
...
             tx_type
         );
-        let hashtype = match tx_type {
-            TransactionType::Unvault => SigHashType::All,
-            TransactionType::Cancel
-            | TransactionType::Emergency
-            | TransactionType::UnvaultEmergency => SigHashType::AllPlusAnyoneCanPay,
-        };
-        if let Err(e) = tx.add_signature(0, pubkey, (sig, hashtype), secp_ctx) {
+        if let Err(e) = tx.add_signature(0, pubkey.key, sig, secp_ctx) {
```
Comment (darosior) [inline src/daemon/sigfetcher.rs:114]: Love to get rid of this hashtype var

**Comment by darosior** [comment]
Did you uncheck "Allow edits from maintainers"? I've troubles updating this PR..

File: src/daemon/control.rs (hunk 1 above)
Comment (danielabrozzoni) [reply src/daemon/control.rs:511]: You modified it again in d23681d58ecb70774ae52fa9637372fbd4a559f2 I think. Or maybe I had troubles with rebasing...

Comment (darosior) [reply src/daemon/control.rs:511]: Should probably be squashed then, it's confusing to have both

**Comment by darosior** [comment]
Can you cherry-pick the first commit in https://github.com/revault/revaultd/pull/251 as the first commit here? In this PR the first commit doesn't pass tests on its own.

**Comment by darosior** [comment]
Can you rebase on master now that i merged #251 ?

**Review by darosior — COMMENTED** [review]
utACK 49610845ec93a28d1172a18a53b5b637c9d23d00 -- thanks for rebasing this!

Will re-kick CI one last time, "just in case" :)

**Pull request merged**